src/server/req_message.c

```diff
@@ -341,7 +342,18 @@ req_relnodesjob(struct batch_request *preq)
 	if ((nodeslist != NULL) && (nodeslist[0] == '\0')) {
 		nodeslist = NULL;
 	}
-	rc = free_sister_vnodes(pjob, nodeslist, msg, LOG_BUF_SIZE, preq);
+
+	if (preq->rq_extend && *preq->rq_extend) {
+		keep_select = strchr(preq->rq_extend, '=');
```

Comment:
This is now the server receiving and processing the release_nodes_job request. We would still need to check to make sure preq->rq_extend has been set to "select=<...>" as the request could come from outside pbs_release_nodes.c and could be arbitrarily set to something else. So I suggest doing:

keep_select = strstr(preq->rq_extend, "select=");
if  keep_select == NULL then fail
otherwise, keep_select += strlen("select=") to get to the value after "select=". May want to put "select" in a MACRO in an include file, so that all reference to "select" should refer to the MACRO. Then one can do: keep_select += (strlen(SELECT_RES) + 1)...